src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph.cs

```diff
@@ -413,6 +443,55 @@ private void ComputedDependencySet(ProjectId project, HashSet<ProjectId> result)
             }
         }
 
+        [Conditional("DEBUG")]
+        private static void ValidateForwardReferences(
+            ImmutableHashSet<ProjectId> projectIds,
+            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> referencesMap)
+        {
+            RoslynDebug.Assert(referencesMap is object);
+
+            Debug.Assert(projectIds.Count >= referencesMap.Count);
+            Debug.Assert(referencesMap.Keys.All(projectIds.Contains));
+
+            foreach (var (_, referencedProjects) in referencesMap)
+            {
+                Debug.Assert(!referencedProjects.IsEmpty, "Unexpected empty value in the forward references map.");
+            }
+        }
+
+        [Conditional("DEBUG")]
+        private static void ValidateReverseReferences(
+            ImmutableHashSet<ProjectId> projectIds,
+            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> referencesMap,
```

Comment:
```suggestion
            ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> forwardReferencesMap,
```

(I got myself confused again)

Reply:
➡️ Included in #41288